app/src/main/java/com/vmware/herald/app/AppDelegate.java

```diff
@@ -63,6 +72,18 @@ public void onCreate() {
         sensor = new SensorArray(getApplicationContext(), payloadDataSupplier);
         // Add appDelegate as listener for detection events for logging and start sensor
         sensor.add(this);
+        // Efficacy Loggers
+        PayloadData payloadData = sensor.payloadData();
+        if (BuildConfig.DEBUG) {
+            sensor.add(new ContactLog(this, "contacts.csv"));
+            sensor.add(new StatisticsLog(this, "statistics.csv",payloadData));
+            sensor.add(new DetectionLog(this,"detection.csv", payloadData));
+            new BatteryLog(this, "battery.csv");
+            if (BLESensorConfiguration.payloadDataUpdateTimeInterval != TimeInterval.never) {
+                sensor.add(new EventTimeIntervalLog(this, "statistics_didRead.csv", payloadData, EventTimeIntervalLog.EventType.read));
+            }
+        }
+        logger.info("DEVICE (payload={},description={})", payloadData.shortName(), SensorArray.deviceDescription);
```

Comment:
Should be left in the SensorArray class. Applications should not create a SensorLogged in the 'Herald' subsystem namespace.

Reply:
Line 86 and 47 I mean.

Reply:
ok.  Moved the log statement back in the following commit.